lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -24,6 +26,21 @@ const FINGER_SIZE_PX = 48;
 // to the finger area tapping on the intended element
 const MAX_ACCEPTABLE_OVERLAP_SCORE_RATIO = 0.25;
 
+/**
+ * Returns a tap target augmented with a bounding rect for quick overlapping
+ * rejections. Rect contains all the client rects, with a width and height of at
+ * least FINGER_SIZE_PX.
+ * @param {LH.Artifacts.TapTarget[]} targets
+ * @return {BoundedTapTarget[]}
+ */
+function getBoundedTapTargets(targets) {
+  return targets.map(tapTarget => {
+    return {
+      tapTarget,
+      boundingRect: getBoundingRectWithMinimimSize(tapTarget.clientRects, FINGER_SIZE_PX),
```

Comment:
We could have a case where a target has two small CRs that won't be merged together by `getTappableRectsFromClientRects`. A finger placed  on the one of the CRs would reach outside the bounding box and we would no longer report a failure when previously we did. `getBoundingRectWithPadding` would be more accurate.

This is super hypothetical though, especially since we allow some finger overlap without failing.

Reply:
> `getBoundingRectWithPadding` would be more accurate.

d'oh, good catch. We should totally do that.

There shouldn't be much more overhead since most cases should still reject (on a normal page with links and buttons, the vast majority of the time any two randomly picked tap targets on the page aren't that near each other), so we should be precise. This also lets us tweak the overlap thresholds or rules later without worrying we're missing cases.

Reply:
yeah, it's about 15% slower on the wikipedia case, but it's still 93% faster than before, so we're still good :)